qiita_pet/handlers/api_proxy/tests/test_sample_template.py

```diff
@@ -0,0 +1,154 @@
+            'tot_nitro': [('1.3', 9), ('1.41', 9), ('1.51', 9)],
+            'depth': [('0.15', 27)],
+            'anonymized_name': [('SKB1', 1), ('SKB2', 1), ('SKB3', 1),
+                                ('SKB4', 1), ('SKB5', 1), ('SKB6', 1),
+                                ('SKB7', 1), ('SKB8', 1), ('SKB9', 1),
+                                ('SKD1', 1), ('SKD2', 1), ('SKD3', 1),
+                                ('SKD4', 1), ('SKD5', 1), ('SKD6', 1),
+                                ('SKD7', 1), ('SKD8', 1), ('SKD9', 1),
+                                ('SKM1', 1), ('SKM2', 1), ('SKM3', 1),
+                                ('SKM4', 1), ('SKM5', 1), ('SKM6', 1),
+                                ('SKM7', 1), ('SKM8', 1), ('SKM9', 1)]},
+               'num_samples': 27}
+        self.assertEqual(obs, exp)
+
+    def test_sample_template_info_no_access(self):
+        obs = sample_template_info(1, 'demo@microbio.me')
+        exp = {'status': 'error',
+               'message': 'User does not have access to study'}
+        self.assertEqual(obs, exp)
+
+    def test_process_sample_template(self):
+        obs = process_sample_template(1, 'test@foo.bar', '16S',
+                                      'uploaded_file.txt')
+        exp = {'status': 'error',
+               'message': 'Empty file passed!',
+               'file': 'uploaded_file.txt'}
+        self.assertEqual(obs, exp)
+
+    def test_process_sample_template_no_access(self):
+        obs = process_sample_template(1, 'demo@microbio.me', '16S', 'filepath')
+        exp = {'status': 'error',
+               'message': 'User does not have access to study'}
+        self.assertEqual(obs, exp)
+
+    def test_update_sample_template(self):
+        obs = update_sample_template(1, 'test@foo.bar', 'uploaded_file.txt')
+        exp = {'status': 'error',
+               'message': 'Empty file passed!',
+               'file': 'uploaded_file.txt'}
+        self.assertEqual(obs, exp)
+
+    def test_update_sample_template_no_access(self):
+        obs = update_sample_template(1, 'demo@microbio.me', 'filepath')
+        exp = {'status': 'error',
+               'message': 'User does not have access to study'}
+        self.assertEqual(obs, exp)
+
+    def test_delete_sample_template(self):
+        obs = delete_sample_template(1, 'test@foo.bar')
+        exp = {'status': 'error',
+               'message': 'Sample template can not be erased because there are'
+                          ' prep templates associated.'}
+        self.assertEqual(obs, exp)
+
+    def test_delete_sample_template_no_access(self):
+        obs = delete_sample_template(1, 'demo@microbio.me')
+        exp = {'status': 'error',
+               'message': 'User does not have access to study'}
+        self.assertEqual(obs, exp)
+
+    def test_get_sample_template_filepaths(self):
+        obs = get_sample_template_filepaths(1, 'test@foo.bar')
+        exp = [(14, '/Users/jshorens/Repositories/qiita/qiita_db/support_'
```

Comment:
Yes. More specifically, the interface should not leak server side details: the client should not care about the nature of how templates are stored especially since the client doesn't have access to the filesystem.